File: src/main/java/um/haberes/core/repository/LegajoInasistenciaRepository.java
package um.haberes.core.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import um.haberes.core.kotlin.model.LegajoInasistencia;

import java.util.Optional;

public interface LegajoInasistenciaRepository extends JpaRepository<LegajoInasistencia, Long> {

    Optional<LegajoInasistencia> findByLegajoIdAndAnhoAndMesAndFacultadIdAndGeograficaId(
            Long legajoId,
            Integer anho,
            Integer mes,
            Integer facultadId,
            Integer geograficaId
    );

}

File: src/main/java/um/haberes/core/kotlin/repository/LegajoInasistenciaRepository.kt
package um.haberes.core.kotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import um.haberes.core.kotlin.model.LegajoInasistencia
import java.util.Optional

interface LegajoInasistenciaRepository : JpaRepository<LegajoInasistencia?, Long?> {

    fun findByLegajoIdAndAnhoAndMesAndFacultadIdAndGeograficaId(
        legajoId: Long,
        anho: Int,
        mes: Int,
        facultadId: Int,
        geograficaId: Int
    ): Optional<LegajoInasistencia?>?;

}